designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/RadioButtonInput.kt

```diff
@@ -0,0 +1,133 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Cancel
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import org.hisp.dhis.mobile.ui.designsystem.component.InputShellState.DISABLED
+import org.hisp.dhis.mobile.ui.designsystem.component.InputShellState.UNFOCUSED
+import org.hisp.dhis.mobile.ui.designsystem.component.Orientation.HORIZONTAL
+import org.hisp.dhis.mobile.ui.designsystem.component.Orientation.VERTICAL
+
+@Composable
+fun RadioButtonInput(
+    title: String,
+    radioButtonData: List<RadioButtonData>,
+    modifier: Modifier = Modifier,
+    orientation: Orientation = VERTICAL,
+    state: InputShellState = UNFOCUSED,
+    supportingText: List<SupportingTextData>? = null,
+    legendData: LegendData? = null,
+    isRequired: Boolean = false,
+    onSelectionChanged: (uid: String) -> Unit,
+    onClearSelection: (() -> Unit)? = null,
+) {
+    InputShell(
+        modifier = modifier.testTag("RADIO_BUTTON_INPUT"),
+        isRequiredField = isRequired,
+        title = title,
+        state = state,
+        legend = {
+            legendData?.let {
+                Legend(legendData, modifier.testTag("RADIO_BUTTON_INPUT_LEGEND"))
+            }
+        },
+        supportingText = {
+            supportingText?.forEach { label ->
+                SupportingText(
+                    label.text,
+                    label.state,
+                    modifier = modifier.testTag("RADIO_BUTTON_INPUT_SUPPORTING_TEXT"),
+                )
+            }
+        },
+        inputField = {
+            when (orientation) {
+                HORIZONTAL -> RadioButtonInputRow(
+                    radioButtonData = radioButtonData,
+                    state = state,
+                    onSelectionChanged = onSelectionChanged,
+                )
+
+                VERTICAL -> RadioButtonInputColumn(
+                    radioButtonData = radioButtonData,
+                    state = state,
+                    onSelectionChanged = onSelectionChanged,
+                )
+            }
+        },
+        primaryButton = {
+            val isClearButtonVisible = radioButtonData.firstOrNull { it.selected } != null && state != DISABLED
+            if (isClearButtonVisible) {
+                IconButton(
+                    modifier = Modifier.testTag("RADIO_BUTTON_INPUT_CLEAR_BUTTON"),
+                    icon = {
+                        Icon(
+                            imageVector = Icons.Outlined.Cancel,
+                            contentDescription = "Icon Button",
+                        )
+                    },
+                    onClick = {
+                        onClearSelection?.invoke()
+                    },
+                )
+            }
+        },
+    )
+}
+
+@Composable
+private fun RadioButtonInputColumn(
```

Comment:
Since we now have the input component for the radio button, we can get remove the `RadioButtonBlock` right? It doesn't looks like we will use that component by it self. cmiiw

Reply:
I believe according to figma design we will use it for other components  like in the bottom sheet 
![image](https://github.com/dhis2/dhis2-mobile-ui/assets/44061143/49b84504-4295-4db8-83b3-f577850263c5)

the input component will make use of the RadioButtonBlock but I don't think it should be in the same file as InputRadioButton as it won't necessarily be contained within the InputShell. I'll ask Marcos and get back to you

Reply:
Shouldn't that be a list of items rather than column then? I think it would be great to expose the content area for the bottom sheet for that and use `LazyColumn` in the app. But we can extract out the entire thing as component if that's needed.

Reply:
ok I've confirmed that we will be using RadioButtonBlock  for different components so if you could use that one for this component that would be great, feel free to adapt it as you see fit